### PR TITLE
Export tree to the dot format

### PR DESCRIPTION
**Pull request by Miguellissimo**
While using the treelib library I've figured out that there was no possibility to export a tree to the dot format of the Graphviz software package. Only a show method is provided. Therefore I have added a function to export the tree to the dot format and created a few unit tests to check if its working correctly.

### CODE REVIEW

**Comment by Natim** [comment]
Your tests are not passing for Python3. Do you need help to make them pass?

**Comment by Miguellissimo** [comment]
Hey Natim, thanks for the feedback. I hope it works now between the different python versions

File: tests/test_treelib.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 import sys
+import os
```
Comment (Natim) [inline tests/test_treelib.py]: nit: we shall try to order import alphabetically and as described in PEP8 - http://legacy.python.org/dev/peps/pep-0008/#imports

**Comment by Natim** [comment]
I like this feature also it may worth making it a plugin for treelib.

ie: A function to which you pass a tree and that build the dotfile? rather than a method of the tree?

**Comment by caesar0301** [comment]
Hi @Miguellissimo, I also think its a good practice to contain your method in a plugin, as suggested by @Natim .  I am very glad to add a new module as `treelib.plugins` to maintain the community contributed extensions for the basic Tree class. Whats your idea?

**Comment by Miguellissimo** [comment]
Hi @caesar0301, yes I think its a good idea. I just didn't know that there is a module for plugins in progress. As far as I can see its not yet pushed to the master branch. Therefore I'll wait till its finally available and move my function to that module.

**Comment by Natim** [comment]
@Miguellissimo I think @caesar0301 was suggesting that you create it with your patch.

**Comment by caesar0301** [comment]
@Miguellissimo I have added the module and you can add your patch there. :+1:

**Comment by Miguellissimo** [comment]
Here we are, I moved the test cases to a another module as well to keep them apart from the tree ones.
I was unsure if there should be a general class for the plugin tests. But due to the case that future plugins  may use a different setup I kept them in their own class.

**Comment by caesar0301** [comment]
Nice job! The general test class for plugins may be required when this repo grows in the future. Thanks!

File: treelib/plugins.py [hunk 2]
```diff
@@ -1,5 +1,38 @@
...
+            f.write('\t' + n + '\n')
+        
+        f.write('\n')
+        for c in connections:
+            f.write('\t' + c + '\n')
+
+        f.write('}')
+
+if __name__ == '__main__':
+    pass
```
Comment (Natim) [inline treelib/plugins.py]: Missing empty line at the end of the file

File: tests/test_plugins.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        self.assertTrue(os.path.isfile('ŕʩϢ.dot'), "The file ŕʩϢ.dot could not be found.")
+        generated = self.read_generated_output('ŕʩϢ.dot')
+        self.assertEqual(expected, generated, "The generated file content is not the expected one")
+        os.remove('ŕʩϢ.dot')
+
+    def tearDown(self):
+        self.tree = None
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (Natim) [inline tests/test_plugins.py]: Missing empty line at the end of the file

File: tests/test_treelib.py (hunk 1 above)
Comment (Natim) [inline tests/test_treelib.py]: Do you finally needs to had those imports?

File: tests/test_treelib.py [hunk 4]
```diff
@@ -2,6 +2,8 @@
...
+import codecs
```
Comment (Natim) [inline tests/test_treelib.py]: Do you finally needs to had those imports?

**Comment by Natim** [comment]
I added some nit comments. Thanks you for this feature.

**Pull request merged**